### ISSUE TITLE
Python 3.10: problems with the modules installation paths

### ISSUE THREAD

**Issue by OscarL** [haikuports/haikuports#7531]
Opening an issue for this so the problem (and its cause) doesn't gets buried...

As mentioned on #7519, Python 3.10 has some issues with the paths it's using for installing modules:

"system" modules can't be installed, as the "site-package" directory it tries to use (under `/system/lib/pyton3.10/`) is read-only.  So it reverts to user installations.

But those user installations are using paths as `~/.local/bin/`, `~/.local/lib/`, instead of the expected `~/config/non-packaged/bin/` and `~/config/non-packaged/lib/`. This makes it impossible to import "successfully" installed modules, and also prevents uninstalling them with pip.

Python 3.7 to 3.9 seem to have the proper paths[1] for installations, and we should carry that to 3.10 and up.

----

[1] Small caveat.. 3.7 to 3.9 seem to have some left-over code looking at `~/.local/bin/`. This is easy to see doing the following:

```bash
> pip3.10 install --user pyserial
> pip3.9 install --user pyserial
> pip3.9 uninstall pyserial
```

The first line "succeeds", but has the problems mentioned above.

The last line is the interesting one, because now pip3.9 offers to delete the following

```
/boot/home/.local/bin/pyserial-miniterm
/boot/home/.local/bin/pyserial-ports
/boot/home/config/non-packaged/bin/pyserial-miniterm
/boot/home/config/non-packaged/bin/pyserial-ports
/boot/home/config/non-packaged/lib/python3.9/site-packages/pyserial-3.5.dist-info/*
/boot/home/config/non-packaged/lib/python3.9/site-packages/serial/*
```

when the first two were installed by pip3.10 and NOT by pip3.9 (3.7 and 3.8 behaves the same here).

It might be worth to investigate this issue too.

**Comment by korli** [comment]
> I accidentally deleted a previous comment where I suggested that we should also patch sysconfig.py's `_getuserbase()` also for Python versions 3.7 to 3.9 in the same way it is done for site.py's `_getuserbase()`.
> 
> Thoughts?

Yeah, both `_getuserbase()` should be synched. Good idea.

**Comment by Begasus** [comment]
Food for thoughts, when using pip to install pyserial, the binaries are installed in `bin`, I'm guessing the latest python installed version will have the binaries available and will overwrite the existing ones?
Won't that lead to conflicts?

**Comment by korli** [comment]
@Begasus does this work on other platforms?

**Comment by Begasus** [comment]
> @Begasus does this work on other platforms?

No idea @korli I only use/check it on Haiku, with the test to install with pip I saw something that reminded me on the lua recipes where the binaries were overwriten with another version so I thought I would check it here.



> TL;DR.: working (or broken? :-P) as intended, by pip standards.

Nice read, and yes, this isn't a python issue, so not realy related here, just wanted to pop up a balloon. ;) (nice discussion upstream though) :D
